pages/workspaces/[workspaceId]/repository-insights/[insightId]/activity.tsx

```diff
@@ -0,0 +1,108 @@
+import { useEffect, useState } from "react";
+import { GetServerSidePropsContext } from "next";
+import { createPagesServerClient } from "@supabase/auth-helpers-nextjs";
+
+import SEO from "layouts/SEO/SEO";
+import fetchSocialCard from "lib/utils/fetch-social-card";
+import getInsightTeamMemberAccess from "lib/utils/get-insight-team-member";
+import { MemberAccess } from "components/molecules/TeamMembersConfig/team-members-config";
+import useInsightRepositories from "lib/hooks/useInsightRepositories";
+import { WorkspaceLayout } from "components/Workspaces/WorkspaceLayout";
+import HubPageLayout from "layouts/hub-page";
+import Activity from "components/organisms/Activity/activity";
+
+interface InsightPageProps {
+  insight: DbUserInsight;
+  ogImage?: string;
+  workspaceId: string;
+}
+
+const HubPage = ({ insight, ogImage, workspaceId }: InsightPageProps) => {
+  const { data: insightRepos } = useInsightRepositories(insight.id);
+  const repositories = insightRepos.map((repo) => repo.repo_id);
+  const [hydrated, setHydrated] = useState(false);
+
+  useEffect(() => {
+    setHydrated(true);
+  }, []);
+
+  if (!hydrated) {
+    return (
+      <>
+        <SEO
+          title={`${insight.name} | Open Sauced Insights `}
+          description={`${insight.name} Insights on OpenSauced`}
+          image={ogImage}
+          twitterCard="summary_large_image"
+        />
+      </>
+    );
+  }
```

Comment:
refactor (blocking): use the `useHasMounted()` hook from `import { useHasMounted } from "lib/hooks/useHasMounted";` Same for other pages/components in the PR.

```suggestion
  const hasMounted = useHasMounted(); 
  
  if (!hasMounted) {
    return (
        <SEO
          title={`${insight.name} | Open Sauced Insights `}
          description={`${insight.name} Insights on OpenSauced`}
          image={ogImage}
          twitterCard="summary_large_image"
        />
    );
  }
```

Reply:
Ok cool. Wasn't aware of this hook